test/jdk/java/awt/dnd/IntraJVMGetDropSuccessTest.java

```diff
@@ -0,0 +1,283 @@
+                System.err.println("WARNING: Couldn't locate " + canvas2);
+                return;
+            }
+
+            final Point dstPoint1 = getCenterLocationOnScreen(canvas1);
+
+            if (!pointInComponent(robot, dstPoint1, canvas1)) {
+                System.err.println("WARNING: Couldn't locate " + canvas1);
+                return;
+            }
+
+            final Point dstPoint2 = getCenterLocationOnScreen(canvas3);
+            if (!pointInComponent(robot, dstPoint2, canvas3)) {
+                System.err.println("WARNING: Couldn't locate " + canvas3);
+                return;
+            }
+
+            test(robot, srcPoint, dstPoint1, false);
+            test(robot, srcPoint, dstPoint2, true);
+            test(robot, srcPoint, dstPoint1, false);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException("The test failed.");
+        }
+    }
+
+    public Point getCenterLocationOnScreen(Component c) throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            p = c.getLocationOnScreen();
+            d = c.getSize();
+        });
+        p.translate(d.width / 2, d.height / 2);
+        return p;
+    }
+
+    public void test(Robot robot, Point src, Point dst, boolean success)
+      throws InterruptedException {
+
+        dragSourceListener.reset();
+        robot.mouseMove(src.x, src.y);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
+
+        for (Point p = new Point(src); !p.equals(dst);
+             p.translate(sign(dst.x - p.x),
+                         sign(dst.y - p.y))) {
+            robot.mouseMove(p.x, p.y);
+            Thread.sleep(50);
+        }
+
+        synchronized (SYNC_LOCK) {
+            robot.mouseRelease(InputEvent.BUTTON1_MASK);
+            SYNC_LOCK.wait();
+        }
+
+        if (!dragSourceListener.isDropFinished()) {
+            throw new RuntimeException("Drop not finished");
+        }
+
+        if (dragSourceListener.getDropSuccess() != success) {
+            throw new RuntimeException("getDropSuccess() returned wrong value:"
+                                       + dragSourceListener.getDropSuccess());
+        }
+    }
+
+    public void reset() {
+        clickedComponent = null;
+    }
+
+    public void eventDispatched(AWTEvent e) {
+        if (e.getID() == MouseEvent.MOUSE_RELEASED) {
+            clickedComponent = (Component)e.getSource();
+            synchronized (SYNC_LOCK) {
+                SYNC_LOCK.notifyAll();
+            }
+        }
+    }
+
+    boolean pointInComponent(Robot robot, Point p, Component comp)
+      throws InterruptedException {
+        robot.waitForIdle();
+        reset();
+        robot.mouseMove(p.x, p.y);
+        robot.mousePress(InputEvent.BUTTON1_MASK);
+        synchronized (SYNC_LOCK) {
+            robot.mouseRelease(InputEvent.BUTTON1_MASK);
+            SYNC_LOCK.wait(MOUSE_RELEASE_TIMEOUT);
+        }
+
+        Component c = clickedComponent;
```

Comment:
this field is used on EDT and the main thread, some synchronisation is needed.

Reply:
Should I put that field on the EDT as well?